src/Personal/actionCreator.js

```
@@ -0,0 +1,144 @@
+  return {
+    type: actionType.signupSuccessful,
+    payload: userInfo,
+  };
+}
+
+export function signupFailed() {
+  return {
+    type: actionType.signupFailed,
+  };
+}
+
+export function restoreStart() {
+  return {
+    type: actionType.RESTORE_START,
+  };
+}
+
+export function restoreSuccessful(payload) {
+  return {
+    type: actionType.RESTORE_SUCCESSFUL,
+    payload,
+  };
+}
+
+export function restoreFailed(payload) {
+  return {
+    type: actionType.RESTORE_FAILED,
+    payload,
+  };
+}
+
+export const requestBackup = () => (dispatch, getState) => {
+  const { transactions } = getState().transactions;
+  if (!accessToken) {
+    dispatch(backupFailed('Unauthorized token'));
+    return;
+  }
+  dispatch(backupStart());
+  fetch(`${BACKEND_URL}/backup`, {
+    method: 'POST',
+    mode: 'cors',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ transactions }),
+  }).then((response) => {
+    if (response.status === 200) {
+      setTimeout(() => dispatch(backupSuccessful(moment().unix())), 300);
+      throw new Error(200);
+    } else {
+      return response.json();
+    }
+  }).then((response) => {
+    setTimeout(() => dispatch(backupFailed(response.message)), 300);
+  }).catch((error) => {
+    if (error.message !== '200') {
+      dispatch(backupFailed('Network error, Please try again later'));
+    }
+  });
+};
+
+export const requestRestore = () => (dispatch) => {
+  if (!accessToken) {
+    dispatch(restoreFailed('Unauthorized token'));
+    return;
+  }
+  dispatch(restoreStart());
+
+  fetch(`${BACKEND_URL}/backup`, {
+    method: 'GET',
+    mode: 'cors',
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+  }).then((response) => {
+    if (response.status === 401) {
+      dispatch(restoreFailed(response.message));
+      throw new Error(response.status);
```

the same as above, you can either throw an error  or dispatch the action, but do not both